File: the-largest-number-finder.py
import math

def largest_of_all(arr):
    results = []
    for sub_arr in arr:
        max_num = sub_arr[0]
        for num in sub_arr[1:]:
            if num > max_num:
                max_num = num
            results.append(max_num)
    return results

#variant

def largest_of_all_pythonic(arr):
    return [max(sub_arr) for sub_arr in arr]

#ex

sub_arr = [13,27,18]

print(max(*sub_arr))

#ex2

fruits = ["🍎", "🍌"]
vegetables = ["🥦", "🥕"]

food_list = [*fruits, *vegetables]

original = [1,2,3]
copy_arr = [*original]

print(copy_arr)

File: the-largest-number-finder.js
function largestOfAll(arr) {
  const results = [];

  for (let i = 0; i < arr.length; i++) {
    let maxNum = arr[i][0];

    for (let j = 1; j < arr[i].length; j++) {
      if (arr[i][j] > maxNum) {
        maxNum = arr[i][j];
      }
    }

    results.push(maxNum);
  }

  return results;
}
//variant 

function largestOfAll(arr) {
    return arr.map(subArr => Math.max(...subArr));
}

//ex

const fruits = ["🍎", "🍌"];
const vegetables = ["🥦", "🥕"];

const foodList = [...fruits, ...vegetables];
console.log(foodList); 

const fullMenu = ["🍞", ...fruits, "🧀", ...vegetables];
console.log(fullMenu);

//

const original = [1,2,3];
let copy = [...original];
copy.push(99);

console.log(original);

console.log(copy);